server/channels/store/sqlstore/access_control_policy_store.go

```diff
@@ -0,0 +1,404 @@
+		return nil, errors.Wrapf(err, "failed to fetch policy with id=%s", policy.ID)
+	}
+
+	if existingPolicy != nil {
+		// move existing policy to history
+		tmp, err2 := fromModel(existingPolicy)
+		if err2 != nil {
+			return nil, errors.Wrapf(err2, "failed to parse policy with id=%s", policy.ID)
+		}
+
+		data := tmp.Data
+		props := tmp.Props
+		if s.IsBinaryParamEnabled() {
+			data = AppendBinaryFlag(data)
+			props = AppendBinaryFlag(props)
+		}
+
+		query := s.getQueryBuilder().
+			Insert("AccessControlPolicyHistory").
+			Columns("ID", "Name", "Type", "CreateAt", "Revision", "Version", "Data", "Props").
+			Values(tmp.ID, tmp.Name, tmp.Type, tmp.CreateAt, tmp.Revision, tmp.Version, data, props)
+
+		_, err = tx.ExecBuilder(query)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to save policy with id=%s to history", policy.ID)
+		}
+
+		err = s.deleteT(rctx, tx, existingPolicy.ID)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to delete policy with id=%s", policy.ID)
+		}
+	}
+
+	preSaveAccessControlPolicy(policy, existingPolicy)
+
+	storePolicy, err := fromModel(policy)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse policy with Id=%s", policy.ID)
+	}
+
+	data := storePolicy.Data
+	props := storePolicy.Props
+	if s.IsBinaryParamEnabled() {
+		data = AppendBinaryFlag(data)
+		props = AppendBinaryFlag(props)
+	}
+
+	query := s.getQueryBuilder().
+		Insert("AccessControlPolicies").
+		Columns(accessControlPolicySliceColumns()...).
+		Values(storePolicy.ID, storePolicy.Name, storePolicy.Type, storePolicy.Active, storePolicy.CreateAt, storePolicy.Revision, storePolicy.Version, data, props)
+
+	_, err = tx.ExecBuilder(query)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to save policy with id=%s", policy.ID)
+	}
+
+	cp, err := storePolicy.toModel()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse policy with id=%s", policy.ID)
+	}
+
+	if err = tx.Commit(); err != nil {
+		return nil, errors.Wrap(err, "commit_transaction")
+	}
+
+	return cp, nil
+}
+
+func (s *SqlAccessControlPolicyStore) Delete(rctx request.CTX, id string) error {
+	tx, err := s.GetMaster().Beginx()
+	if err != nil {
+		return errors.Wrap(err, "failed to start transaction")
+	}
+	defer finalizeTransactionX(tx, &err)
+
+	existingPolicy, err := s.getT(rctx, tx, id)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return errors.Wrapf(err, "failed to fetch policy with id=%s", id)
+	}
+
+	if existingPolicy != nil {
+		tmp, err2 := fromModel(existingPolicy)
+		if err2 != nil {
+			return errors.Wrapf(err2, "failed to parse policy with id=%s", id)
+		}
+		data := tmp.Data
+		props := tmp.Props
+		if s.IsBinaryParamEnabled() {
+			data = AppendBinaryFlag(data)
+			props = AppendBinaryFlag(props)
+		}
+
+		query := s.getQueryBuilder().
+			Insert("AccessControlPolicyHistory").
+			Columns("ID", "Name", "Type", "CreateAt", "Revision", "Version", "Data", "Props").
```

Comment:
Can we create a function for this similar to `accessControlPolicySliceColumns`? Its being used in two places.